### PR TITLE
Add a definition for react-modal-v3.11.x

### PR DESCRIPTION
**Pull request by knu**
- Links to documentation: http://reactcommunity.org/react-modal/
- Link to GitHub or NPM: https://github.com/reactjs/react-modal
- Type of contribution: new definition for a new version

### CODE REVIEW

**Review by AndrewSouthpaw — COMMENTED** [review]
Nice work, thanks for putting this together!

Can you go through the build errors in Travis and address those as well?

**Comment by knu** [comment]
Fixed.  I ignored the error because it seemed recent PRs got merged without fixing warnings and I thought those warnings are OK.  Good to know there was a decent fix.

File: definitions/npm/react-modal_v3.11.x/flow_v0.104.x-/react-modal_v3.11.x.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    onAfterOpen?: ({ overlayEl: HTMLElement, contentEl: HTMLElement, ... }) => void | Promise<void>,
+    onAfterClose?: () => void | Promise<void>,
+    onRequestClose?: (SyntheticEvent<>) => void,
+    aria?: { [key: string]: string, ... },
+    role?: string,
+    contentLabel?: string,
+    overlayRef?: (node: ?HTMLElement) => mixed,
+    containerRef?: (node: ?HTMLElement) => mixed,
+    ...
```
Comment (Brianzchen) [inline definitions/npm/react-modal_v3.11.x/flow_v0.104.x-/react-modal_v3.11.x.js:39]: Should we explicitly add a `children` prop too?

File: definitions/npm/react-modal_v3.11.x/flow_v0.54.1-v0.103.x/react-modal_v3.11.x.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+declare module 'react-modal' {
+  declare type ModalProps = {
+    isOpen?: boolean,
```
Comment (Brianzchen) [inline definitions/npm/react-modal_v3.11.x/flow_v0.54.1-v0.103.x/react-modal_v3.11.x.js:3]: > The only required prop for the modal object is isOpen, which indicates whether the modal should be displayed
```suggestion
    isOpen: boolean,
```

File: definitions/npm/react-modal_v3.11.x/flow_v0.54.1-v0.103.x/react-modal_v3.11.x.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    onAfterClose?: () => void | Promise<void>,
+    onRequestClose?: (SyntheticEvent<>) => void,
+    aria?: {
+      [key: string]: string
+    },
+    role?: string,
+    contentLabel?: string,
+    overlayRef?: (node: ?HTMLElement) => mixed,
+    containerRef?: (node: ?HTMLElement) => mixed,
+  };
```
Comment (Brianzchen) [inline definitions/npm/react-modal_v3.11.x/flow_v0.54.1-v0.103.x/react-modal_v3.11.x.js:42]: Should allow spreading extra props
```suggestion
    ...
  };
```

File: definitions/npm/react-modal_v3.11.x/flow_v0.104.x-/react-modal_v3.11.x.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+declare module 'react-modal' {
+  declare type ModalProps = {
+    isOpen?: boolean,
```
Comment (Brianzchen) [inline definitions/npm/react-modal_v3.11.x/flow_v0.104.x-/react-modal_v3.11.x.js:3]:
```suggestion
    isOpen: boolean,
```

File: definitions/npm/react-modal_v3.11.x/flow_v0.104.x-/react-modal_v3.11.x.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      ...
+    },
+    onAfterOpen?: ({ overlayEl: HTMLElement, contentEl: HTMLElement, ... }) => void | Promise<void>,
+    onAfterClose?: () => void | Promise<void>,
+    onRequestClose?: (SyntheticEvent<>) => void,
+    aria?: { [key: string]: string, ... },
+    role?: string,
+    contentLabel?: string,
+    overlayRef?: (node: ?HTMLElement) => mixed,
+    containerRef?: (node: ?HTMLElement) => mixed,
```
Comment (Brianzchen) [inline definitions/npm/react-modal_v3.11.x/flow_v0.104.x-/react-modal_v3.11.x.js:38]: missing `contentRef` I think?

**Comment by AndrewSouthpaw** [comment]
Thanks for that review, @Brianzchen ! In the spirit of moving forward, how about I merge this in, and you can throw in those corrections into a separate PR?

**Pull request merged**